cloud/misc/proxmox_kvm.py

```diff
@@ -0,0 +1,1020 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Abdoul Bah (@helldorado) <abdoul.bah at alterway.fr>
+
+"""
+Ansible module to manage Qemu(KVM) instance in Proxmox VE cluster.
+This module is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+This software is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this software.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+DOCUMENTATION = '''
+---
+module: proxmox_kvm
+short_description: Management of Qemu(KVM) Virtual Machines in Proxmox VE cluster.
+description:
+  - Allows you to create/delete/stop Qemu(KVM) Virtual Machines in Proxmox VE cluster.
+version_added: "2.3"
+author: "Abdoul Bah (@helldorado) <abdoul.bah at alterway.fr>"
+options:
+  acpi:
+    description:
+      - enable/disable ACPI
+    required: false
+    default: "yes"
+    choices: [ "yes", "no" ]
+    type: boolean
+  agent:
+    description:
+      - enable/disable Qemu GuestAgent
+    required: false
+    default: null
+    choices: [ "yes", "no" ]
+    type: boolean
+  args:
+    description:
+      - pass arbitrary arguments to kvm
+    notes: this option is for experts only
+    default: "-serial unix:/var/run/qemu-server/VMID.serial,server,nowait"
+    required: false
+    type: string
+  api_host:
+    description:
+      - the host of the Proxmox VE cluster
+    required: true
+  api_user:
+    description:
+      - the user to authenticate with
+    required: true
+  api_password:
+    description:
+      - the password to authenticate with
+      - you can use PROXMOX_PASSWORD environment variable
+    default: null
+    required: false
+  autostart:
+    description:
+      - automatic restart after crash (currently ignored)
```

Comment:
@gundalow thank's for all your feedback, @morph027 PR have done some of your preconisation.

This option is ignored by proxmox's API. I suppose it will be working in a future release of proxmox or of it's API, and this won't need any update on this module.